### ISSUE TITLE
I want YOLOv12 to be stronger than YOLOv11.

### ISSUE THREAD

**Issue by Taoboan1999** [ultralytics/ultralytics#21042]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hello, I trained and tested the same training parameters on YOLOv12 and YOLOv11, but why is YOLOv12 less effective than YOLOv11? How can I further optimize it? I want YOLOv12 to be stronger than YOLOv11.

                lr0=0.001, 
                lrf=0.01,  
                momentum=0.937,  
                weight_decay=0.0005,  
                warmup_epochs=3.0,  
                warmup_momentum=0.8, 
                warmup_bias_lr=0.1, 
                box=5.0,  
                cls=0.025, 
                dfl=1.0,
                val=True, 
                plots=True, 
                conf=0.001,
                iou=0.7,  
                hsv_h=0.0, 
                hsv_s=0.0,  
                hsv_v=0.0,  
                degrees=90.0, 
                translate=0.0,  
                scale=0.0,  
                shear=0.0, 
                perspective=0.0,  
                flipud=0.0, 
                fliplr=0.5,  
                bgr=0.0, 
                mosaic=1.0, 
                mixup=0.1,  
                cutmix=0.5,

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Taoboan1999, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response and an Ultralytics engineer will assist you soon 😊

**Comment by glenn-jocher** [comment]
Thanks for your question! YOLO12 is actually **not recommended** for any use cases. YOLO12 suffers from training instability, excessive memory consumption, and slow speed (2-3x slower on CPU) due to the complex attention layers it uses. This explains why you're seeing worse performance compared to YOLO11.

We recommend using [YOLO11](https://docs.ultralytics.com/models/yolo11/) instead, which is our current state-of-the-art model delivering superior performance across all tasks. YOLO11 provides better accuracy, efficiency, and stability compared to YOLO12. For optimal results, stick with YOLO11 and consider using our [hyperparameter tuning guide](https://docs.ultralytics.com/guides/hyperparameter-tuning/) to optimize your training parameters.
